api4/channel.go

```diff
@@ -46,30 +46,30 @@ func (api *API) InitChannel() {
+	api.BaseRoutes.Channel.Handle("/timezones", api.APISessionRequired(getChannelMembersTimezones, model.ScopeChannelsRead, model.ScopeUsersRead)).Methods("GET")
 	api.BaseRoutes.Channel.Handle("/members_minus_group_members", api.APISessionRequired(channelMembersMinusGroupMembers)).Methods("GET")
-	api.BaseRoutes.Channel.Handle("/move", api.APISessionRequired(moveChannel)).Methods("POST")
+	api.BaseRoutes.Channel.Handle("/move", api.APISessionRequired(moveChannel, model.ScopeChannelsUpdate)).Methods("POST")
 	api.BaseRoutes.Channel.Handle("/member_counts_by_group", api.APISessionRequired(channelMemberCountsByGroup)).Methods("GET")
 
-	api.BaseRoutes.ChannelForUser.Handle("/unread", api.APISessionRequired(getChannelUnread)).Methods("GET")
+	api.BaseRoutes.ChannelForUser.Handle("/unread", api.APISessionRequired(getChannelUnread, model.ScopeChannelsRead)).Methods("GET")
 
-	api.BaseRoutes.ChannelByName.Handle("", api.APISessionRequired(getChannelByName)).Methods("GET")
-	api.BaseRoutes.ChannelByNameForTeamName.Handle("", api.APISessionRequired(getChannelByNameForTeamName)).Methods("GET")
+	api.BaseRoutes.ChannelByName.Handle("", api.APISessionRequired(getChannelByName, model.ScopeChannelsRead)).Methods("GET")
+	api.BaseRoutes.ChannelByNameForTeamName.Handle("", api.APISessionRequired(getChannelByNameForTeamName, model.ScopeChannelsRead, model.ScopeTeamsRead)).Methods("GET")
 
-	api.BaseRoutes.ChannelMembers.Handle("", api.APISessionRequired(getChannelMembers)).Methods("GET")
-	api.BaseRoutes.ChannelMembers.Handle("/ids", api.APISessionRequired(getChannelMembersByIds)).Methods("POST")
-	api.BaseRoutes.ChannelMembers.Handle("", api.APISessionRequired(addChannelMember)).Methods("POST")
-	api.BaseRoutes.ChannelMembersForUser.Handle("", api.APISessionRequired(getChannelMembersForTeamForUser)).Methods("GET")
-	api.BaseRoutes.ChannelMember.Handle("", api.APISessionRequired(getChannelMember)).Methods("GET")
-	api.BaseRoutes.ChannelMember.Handle("", api.APISessionRequired(removeChannelMember)).Methods("DELETE")
+	api.BaseRoutes.ChannelMembers.Handle("", api.APISessionRequired(getChannelMembers, model.ScopeChannelsRead, model.ScopeUsersRead)).Methods("GET")
+	api.BaseRoutes.ChannelMembers.Handle("/ids", api.APISessionRequired(getChannelMembersByIds, model.ScopeChannelsRead, model.ScopeUsersRead)).Methods("POST")
+	api.BaseRoutes.ChannelMembers.Handle("", api.APISessionRequired(addChannelMember, model.ScopeChannelsJoin)).Methods("POST")
+	api.BaseRoutes.ChannelMembersForUser.Handle("", api.APISessionRequired(getChannelMembersForTeamForUser, model.ScopeChannelsRead, model.ScopeTeamsRead, model.ScopeUsersRead)).Methods("GET")
+	api.BaseRoutes.ChannelMember.Handle("", api.APISessionRequired(getChannelMember, model.ScopeChannelsRead, model.ScopeUsersRead)).Methods("GET")
+	api.BaseRoutes.ChannelMember.Handle("", api.APISessionRequired(removeChannelMember, model.ScopeChannelsJoin, model.ScopeUsersUpdate)).Methods("DELETE")
```

Comment:
That's not the intention of scopes; they merely control access to types of operation.

1/5 the concept of "self" vs "someone else" is worth considering, but I am a little skeptical that it can be sufficiently isolated, and relying on the user/session actual roles/permissions in Mattermost makes more sense.